platform-hub-api/app/models/kubernetes_namespace.rb

```diff
@@ -0,0 +1,40 @@
+class KubernetesNamespace < ApplicationRecord
+
+  NAME_REGEX = /\A[a-z0-9]([-a-z0-9]*[a-z0-9])?\z/
+
+  include Audited
+
+  audited descriptor_field: :name, associated_field: :service
+
+  belongs_to :service, -> { readonly }
+  belongs_to :cluster, -> { readonly }, class_name: KubernetesCluster
+
+  scope :by_service, ->(s) { where(service: s) }
+  scope :by_cluster, ->(c) { where(cluster: c) }
+
+  validates :name,
+    format: {
+      with: NAME_REGEX,
+      message: "should consist of lowercase letters, numbers and dashes, and must start and end with an alphanumeric character"
+    },
+    uniqueness: {
+      scope: :cluster_id,
+      message: "already exists for the cluster"
+    }
```

Comment:
Should we also validate description presence?

Reply:
I was thinking we make description optional for namespaces. Seems like the namespace name alone is often descriptive enough, so requiring a description would just make people put the same name as the description.